state-chain/custom-rpc/src/lib.rs

```diff
@@ -280,6 +280,21 @@ pub struct RpcEnvironment {
 	pools: PoolsEnvironment,
 }
 
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+pub struct PoolPriceV2 {
+	pub base_asset: RpcAsset,
+	pub quote_asset: RpcAsset,
+	pub buy_price: Option<PoolPriceV1>,
+	pub sell_price: Option<PoolPriceV1>,
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+pub struct RpcPrewitnessedSwap {
+	pub from_asset: RpcAsset,
+	pub to_asset: RpcAsset,
+	pub amounts: Vec<NumberOrHex>,
```

Comment:
Is this the only place you added NumberOrHex?

Reply:
We should actually use `U256`, we are going to stop using NumberOrHex because it serializes strangely.

Reply:
Particularly depending on the size of the value, the serialization is different, which makes LPs lives pointlessly difficult.

Reply:
Ok no problem. 
There's lots of other places that uses `NumberOrHex` to represent AssetAmounts. I'll change this to u256 here